src/app/logic/GameProcessor.ts

```diff
@@ -0,0 +1,114 @@
+import { AnimalNames } from '~src/Enums/AnimalNamesEnum';
+import { Player } from '~src/Player';
+import { Game } from './Game';
+
+export class GameProcessor {
+  constructor(private game: Game) {}
+
+  /**
+   * Starts the timer for current players turn.
+   * When the time is over, resets the timer, sets current player to the next player and starts the timer.
+   * @param turnTimeOverCallback to be executed when the time is over
+   * @param updateRemainingTimeCallback to be executed when the remaining time changes
+   */
+  startTurn(
+    turnTimeOverCallback: (currentPlayer: Player) => void,
+    updateRemainingTimeCallback: (remainingTime: number) => void,
+  ): void {
+    this.game.theTimer.countdown();
+    const turnTimer = setInterval(() => {
+      if (!this.game.theTimer.running) {
+        clearInterval(turnTimer);
+        if (Math.ceil(this.game.theTimer.theTurnTimeLeft) === 0) {
+          turnTimeOverCallback(this.game.theCurrentPlayer);
+
+          this.nextPlayer();
+          this.startTurn(
+            turnTimeOverCallback,
+            updateRemainingTimeCallback,
+          );
+        }
+      }
+      updateRemainingTimeCallback(
+        Math.round(this.game.theTimer.theTurnTimeLeft),
+      );
+    }, 10);
+  }
+
+  /**
+   * Stops the timer for the current player.
+   */
+  stopTurn(): void {
+    this.game.theTimer.resetTurn();
+  }
```

Comment:
In GameController:
`stopTurn(): void {`
`    this.gameProcessor.stopTurn();`
  `}`. 
  Is there a way we could get rid of extra lines?